cmd/secrets-store-csi-driver/main.go

```diff
@@ -147,6 +151,12 @@ func main() {
 	providerClients := secretsstore.NewPluginClientBuilder(*providerVolumePath)
 	defer providerClients.Cleanup()
 
+	// enable provider health check
+	if *providerHealthCheck {
+		klog.InfoS("provider health check enabled", "interval", *providerHealthCheckInterval)
+		go providerClients.EnableHealthCheck(*providerHealthCheckInterval, ctx.Done())
```

Comment:
maybe just pass in the context as a parameter instead of `ctx.Done()` so that the context can also be used within `EnableHealthCheck` for each Version() rpc.

Reply:
The context in main is the context created with background + signal handlers. For the `HealthCheck` we should use a smaller context that terminates sooner for fail-fast instead of this context which will only cancel when pod is terminating. WDYT?

Reply:
Unless we use the `ctx` from here as parent context for the `context.WithCancel` in healthcheck

Reply:
yeah basically by passing in the server context we can use that as the parent instead of `context.Background()` _and_ we'll pickup the cancellation signals, etc automatically

Reply:
sgtm